quotalib/qutoalib.go

```diff
@@ -3,110 +3,190 @@
 	//Function implementation
-	 resp, err := http.Get(fmt.Sprintf("http://172.31.44.22:5050/quota/%s",role))
+	resp, err := http.Get(fmt.Sprintf("http://%s/quota/%s", dc.Endpoint, role))
+
+	if err != nil {
+		log.Printf("Unable to reach the Master error = %v", err)
+		return guarante,nil	
+	}
 
-        if err != nil {
+	body,_ := ioutil.ReadAll(resp.Body)
 
-                log.Printf("Unable to reach the Master error = %v", err)
-                //return
-        }
-        defer resp.Body.Close()
+	defer resp.Body.Close()
 
-        body, err := ioutil.ReadAll(resp.Body)
 
-        if err != nil {
-                log.Printf("Unable to read the body error = %v", err)
-                //return
-        }
-	json_data = []byte(body)
+	error := json.Unmarshal([]byte(body), &data)
 
-	 json.Unmarshal(json_data,&data)
+	if error != nil {
+		log.Printf("Json Unmarshall error = %v", err)
+	}
 
-        if err != nil {
-                log.Printf("Json Unmarshall error = %v", err)
-                //return
-        }
-        log.Println("Get: ",data["infos"])
+	for key, _ := range data.Infos {
+		if data.Infos[key].Role == role {
+			guarante = data.Infos[key].Guarantee
+			log.Println(guarante)
+			break
+		} else if data.Infos[key].Role != role {
+			log.Println("the quota for the", role, " doesn't exist", guarante)
+		}
+	}
 
-	return json_data, nil
+	return guarante, nil
 }
 
 //RemainingResources
 //role: Role name, usually Federation
 //This should automatically resolve the Masters Endpoint from the configuration
 //Retruns how much of the gaurentee has been satisfied for a given role
 //If a role has a quota of 50 CPUs and out of that 35 have been used
-func RemainingResource(role string) (float64, float64, float64, error) {
+func RemainingResource(dc typ.DC, role string) (float64, float64, float64, error) {
 
 	var CPU, MEM, DISK float64
+	var uCPU, uMEM, uDISK float64
+	var qCPU, qMEM, qDISK float64
+
+	var mState typ.MasterState
 
 	//Step 1: This should first call GetQuota() and interpret gaurantee
+	quota, _ := GetQuota(dc, role)
+
+	log.Println("the quota is ", quota, "\n")
+
+	for index, _ := range quota {
+		if quota[index].Name == "cpus" {
+			qCPU = quota[index].Scalar.Value
+		}
+		if quota[index].Name == "mem" {
+			qMEM = quota[index].Scalar.Value
+		}
+		if quota[index].Name == "disk" {
+			qDISK = quota[index].Scalar.Value
+		}
+	}
 
 	//Step 2: This should next call the /state-summary endpoint from the master and figure out all the remaining resources
+	resp, err := http.Get(fmt.Sprintf("http://%s/state.json", dc.Endpoint))
+
+	if err != nil {
+		log.Printf("Unable to reach the Master error = %v", err)
+		return CPU,MEM,DISK,nil
+	}
+
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	defer resp.Body.Close()
+
+	err = json.Unmarshal(body, &mState)
+	if err != nil {
+		log.Printf("Json Unmarshall error = %v", err)
+	}
+
+	for key, _ := range mState.Frameworks {
+		if mState.Frameworks[key].Role == role {
```

Comment:
`mState.Frameworks` is an Array not a MAP, you dont get a key but index.  When you `range` over an array you get Index and the Item. 
you should do this as
```
for _, fW := range mState.Frameworks {
   if fW.Role == role
```
instead of 

```
for key, _ := range mState.Frameworks {
     if mState.Frameworks[key].Role == role {
```